boards/feather_m0/src/lib.rs

```diff
@@ -107,9 +107,17 @@ define_pins!(
     pin flash_sck = a9,
 
     /// SPI MOSI for the external flash
-    #[cfg(all(feature = "express", not(feature = "rfm")))]
+    #[cfg(all(feature = "express", not(feature = "rfm"), not(feature = "adalogger")))]
     pin flash_mosi = a8,
 
+    /// SD card SPI chip select
+    #[cfg(all(feature = "adalogger", not(feature = "rfm"), not(feature = "express")))]
+    pin sd_cs = a8,
+
+    /// SD card detect
+    #[cfg(all(feature = "adalogger", not(feature = "rfm"), not(feature = "express")))]
+    pin sd_cd = a21,
+
     /// SPI MISO for the external flash
```

Comment:
I meant to put that comment here

Reply:
hey @bradleyharden , I tried this out and couldn't quite figure out how to get it to work. I tried:
```rust
use gpio::v2::{AnyPin, self as gpiov2};

#[cfg(feature = "unproven")]
hal::bsp_pins!(
    /// AREF pin - has 1uF capacitor to ground
    PA3 {
        name: aref,
    },
   // (...)
```

and I get
```rust
error: no rules expected the token `}`
   --> src/lib.rs:139:9
    |
139 |         },
    |         ^ no rules expected this token in macro call
```

I also tried putting in an empty alias block since I don't need it for most pins, and it still didn't work. Is this because `gpio` is re-exported and already is valid in `lib.rs`?

As for if I think it should fulfill the needs of having multiple aliases for some overlapping pins, it certainly looks like it.

Reply:
I'm on mobile right now, but I see two things so far. You don't need a comma after each pin {} block. And it's PA03 in gpio::v2. That being said, neither of those directly corresponds to your error. I'll have to look more closely later.

Reply:
Alright I've gotten it to compile a bit farther. Now with this code:
```rust
#[cfg(feature = "unproven")]
hal::bsp_pins!(
    PA03 {
        name: aref,
        aliases: {
            PushPull: ARef
        }
    }

    PA02 {
        name: a0,
        aliases: {
            PushPull: A0
        }
    }
);
```

I'm getting this error:
```rust
error[E0277]: the trait bound `hal::gpio::v2::PushPull: PinMode` is not satisfied
   --> src/lib.rs:133:1
    |
133 | / hal::bsp_pins!(
134 | |     PA03 {
135 | |         name: aref,
136 | |         aliases: {
...   |
146 | |     }
147 | | );
    | |__^ the trait `PinMode` is not implemented for `hal::gpio::v2::PushPull`
```

Reply:
So `PushPull` is an `OutputConfig`, not a `PinMode`. `Output<PushPull>` would be a `PinMode`. Alternatively, you can use the type alias `PushPullOutput`. See the docs [here](https://atsamd-rs.github.io/docs/samd21g/thumbv6m-none-eabi/doc/atsamd_hal/common/gpio/v2/pin/index.html#types) for a list of the `PinMode` type aliases.

Reply:
I'll see if I can make the comma optional, and I'll try to flesh out the docs a little bit more. Thanks for taking a look. Let me know if that change doesn't fix the problem.

Reply:
I've got it compiling now, now just trying to finish it out and convert the example to see how it feels. The SD IO I care about is:
```rust
    let sd_cd = pins.sd_cd.into_pull_up_input(&mut pins.port);
    let mut sd_cs = pins.sd_cs.into_open_drain_output(&mut pins.port);
```
I see `Input<PullUp>`, but no option for an open drain output. Does GPIOv2 support open drain output?

Reply:
Everyone always asks that, haha

SAMD chips don't actually have open drain outputs. They only have push-pull. That was a mistake in the v1 module. I think it must have been copypasta from another chip.

Maybe I should add a section to the docs on the things fixed in v2. That would probably be part of what I still owe @jaredwolff in terms of migration docs.